File: src/main/java/minecraft/rhanjie/locksystem/utility/Utility.java
package minecraft.rhanjie.locksystem.utility;

import org.bukkit.Location;
import org.bukkit.block.Chest;
import org.bukkit.block.DoubleChest;
import org.bukkit.block.data.type.Door;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.inventory.InventoryHolder;

import java.util.List;

public class Utility {
    public static int checkIfElementIsAvailable(FileConfiguration config, String findingPhrase, String configId) {
        List<String> elements = config.getStringList(configId);

        int number = 1;
        for (String element : elements) {
            if (findingPhrase.equalsIgnoreCase(element))
                return number;

            number += 1;
        }

        return -1;
    }

    public static Location getChestSecondPartLocation(Chest chest) {
        InventoryHolder holder = chest.getInventory().getHolder();
        if (holder instanceof DoubleChest) {
            DoubleChest doubleChest = (DoubleChest) holder;
            Chest secondPart = (Chest) doubleChest.getLeftSide();

            if (doubleChest.getLeftSide() == null && doubleChest.getRightSide() == null)
                return null;

            if (chest.getBlock().getLocation().equals(secondPart.getLocation()))
                secondPart = (Chest) doubleChest.getRightSide();

            return secondPart.getBlock().getLocation();
        }

        return null;
    }

    public static Location getDoorSecondPartLocation(Door door, Location secondLocation) {
        String doorPart = door.getHalf().toString();

        if (doorPart.equals("TOP"))
            secondLocation.setY(secondLocation.getBlockY() - 1);

        else if (doorPart.equals("BOTTOM"))
            secondLocation.setY(secondLocation.getBlockY() + 1);

        return secondLocation;
    }
}

File: src/main/kotlin/minecraft.rhanjie.locksystem/utility/Utility.kt
package minecraft.rhanjie.locksystem.utility

import org.bukkit.Location
import org.bukkit.block.Chest
import org.bukkit.block.DoubleChest
import org.bukkit.block.data.type.Door
import org.bukkit.configuration.file.FileConfiguration
import org.bukkit.inventory.InventoryHolder

object Utility {
    fun checkIfElementIsAvailable(config: FileConfiguration, findingPhrase: String, configId: String): Int {
        val elements: List<String> = config.getStringList(configId)

        var number = 1
        for (element in elements) {
            if (findingPhrase.equals(element, true)) {
                return number
            }

            number += 1
        }
        
        return -1
    }

    fun getChestSecondPartLocation(chest: Chest): Location? {
        val holder: InventoryHolder = chest.inventory.holder ?: return null
        if (holder is DoubleChest) {
            val doubleChest: DoubleChest = holder
            var secondPart: Chest = doubleChest.leftSide as Chest

            if (doubleChest.leftSide == null && doubleChest.rightSide == null) {
                return null
            }

            if (chest.block.location.equals(secondPart.location)) {
                secondPart = doubleChest.rightSide as Chest
            }

            return secondPart.block.location
        }

        return null
    }

    fun getDoorSecondPartLocation(door: Door, secondLocation: Location): Location {
        val doorPart: String = door.half.toString()
        if (doorPart == "TOP") {
            secondLocation.y = (secondLocation.blockY - 1).toDouble()
        }

        else if (doorPart == "BOTTOM") {
            secondLocation.y = (secondLocation.blockY + 1).toDouble()
        }

        return secondLocation
    }
}